wagtail/admin/forms/account.py

```diff
@@ -115,7 +122,55 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self._original_avatar = self.instance.avatar
 
+    @contextmanager
+    def create_avatar_context(self, avatar):
+        avatar_context = willow.Image.open(avatar)
+        yield avatar_context
+
+    def reduce_avatar(self, avatar):
+        default_avatar_size = (400, 400)
+
+        with self.create_avatar_context(avatar) as willow_image:
+            width, height = willow_image.get_size()
+
+            should_reduce_avatar = (
+                width > default_avatar_size[0] or height > default_avatar_size[1]
+            )
+
+            if should_reduce_avatar:
+                temp_buffer = BytesIO()
+                original_format = (
+                    willow_image.format_name or willow_image.name.split(".")[-1].lower()
+                )
+                quality = getattr(settings, "WAGTAILIMAGES_AVIF_QUALITY", 80)
+
+                transform = ImageTransform(size=default_avatar_size)
+                operation = ScaleToPercentOperation(None, default_avatar_size[0])
+                transform = operation.run(transform, avatar)
+                clone_img = willow_image.resize(transform.size)
```

Comment:
In hindsight, I suspect calling `willow_image.resize` manually, and avoiding the image transform pipeline might be better, given the operation is quite so simple.